src/savegame.h

```diff
@@ -0,0 +1,229 @@
+#ifndef SAVEGAME_H_
+#define SAVEGAME_H_
+
+#include "stream.h"
+#include <vector>
+
+const int HEROES_DATA_OFFSET 		= 0x019a9b;
+const int HERO_COUNT 				= 67;
+const int HERO_RECORD_SIZE 			= 0x3b;
+const int HERO_NAME_SIZE 			= 0x0f;
+const int HERO_TITLE_SIZE 			= 0x14;
+const int HERO_TYPE_OFFSET 			= 0x24;
+
+const int STARS_COUNT_DATA_OFFSET 	= 0x017ad1;
+const int STARS_DATA_OFFSET 		= 0x017ad3;
+const int STARS_RECORD_SIZE 		= 113;
+const int STARS_NAME_SIZE           = 14;
+const int STARS_SPECIAL_OFFSET      = 0x28;
+const int STARS_VISITED_OFFSET      = 0x33;
+const int STARS_INDICTOR_OFFSET		= 0x39;
+const int STARS_ARTEMIS_OFFSET      = 0x3f;
+
+const int PLAYERS_DATA_OFFSET 		= 0x01aa0f;
+const int PLAYER_RACEPICK_OFFSET 	= 0x000321;
+
+const int TECHNOLOGIES_OFFSET	 	= 0x000325;
+
+const int PLAYER_COUNT 				= 0x08;
+const int PLAYER_NAME_SIZE 			= 0x14;
+const int PLAYER_RACE_SIZE 			= 0x10;
+
+const int MAX_PLAYERS				= 8;
+const int MAX_TECHNOLOGIES 			= 0xcb;
+const int MAX_STARS					= 72;
+
+struct GameConfig {
+    char saveGameName[23];
+    uint16_t stardate;
```

Comment:
You can use simple `int` and `unsigned` for these structure members. The actual field size doesn't matter except when reading/writing a save file.